src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/Writer/CborWriter.Simple.netcoreapp.cs

```diff
@@ -21,7 +21,14 @@ public void WriteHalf(Half value)
         {
             EnsureWriteCapacity(1 + sizeof(short));
             WriteInitialByte(new CborInitialByte(CborMajorType.Simple, CborAdditionalInfo.Additional16BitData));
-            BinaryPrimitives.WriteHalfBigEndian(_buffer.AsSpan(_offset), value);
+            if (Half.IsNaN(value) && !CborConformanceModeHelpers.RequiresPreservingFloatPrecision(ConformanceMode))
```

Comment:
I'm not sure I understand why the second check is necessary. It is meant to check if the current conformance mode permits using lower-precision representation, however `Half` is the lowest one available per the spec.

Reply:
Half NaN may also have a sign and payload and CTAP2 mode description says *representations of any floating-point values are not changed*. So I opted for writing the input verbatim.